src/tokenizer.jl

```diff
@@ -61,7 +61,84 @@ function Tokenizer(tokenizer_path::String, vocab_size::Int)
     return Tokenizer(tokens, vocab_scores)
 end
 
-const BOS_TOKEN::Int32 = 1
+const BOS_TOKEN::Int32 = 2
+const EOS_TOKEN::Int32 = 3
+
+"""
+Encode the string text (input) into an upper-bound preallocated token array
+bos != 0 means prepend the BOS token (=1), eos != 0 means append the EOS token (=2)
+
+llama2.c correspondence: encode (l. 452)
+"""
+function encode(tokenizer::Tokenizer, text::String)
+    # stores merge candidates of two consecutive tokens
+    tokens::Vector{Int} = []
+
+    # if the text is empty there is no encoding so we can return the empty tokens array
+    if isempty(text)
+        return tokens
+    end
```

Comment:
This is not in the original C code, better leave it out. When an empty string is provided, the return value should not be an empty array but an array with a single element - the BOS token.